lib/elastic_record/index/bulk_documents.rb

```diff
@@ -0,0 +1,83 @@
+require 'active_support/core_ext/object/to_query'
+
+module ElasticRecord
+  class Index
+    module BulkDocuments
+      def index_document(id, document, doctype: model.doctype, parent: nil, index_name: alias_name)
+        if batch = current_bulk_batch
+          instructions = { _index: index_name, _type: doctype.name, _id: id }
```

Comment:
Add similar ID verification for nil?

Reply:
And retry?

Reply:
`id` is not required. Id is optional with the Index Document API.

`retry_on_conflict` is only available with the Update API.